debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/JdbcChangeEventSink.java

```diff
@@ -88,14 +102,20 @@ public void execute(Collection<SinkRecord> records) {
                 continue;
             }
 
+            final CollectionId collectionId = optionalCollectionId.get();
+
             if (record.isTombstone()) {
```

Comment:
@msillence Should we maybe move the below code for `if (record.isDelete()) { ... }` up here? `record.isDelete()` covers the case of tombstones and Debezium delete change event records. The code below can be deleted.

Mind that this new behavior will cause 2 deletes executed because `tombstones.on.delete` in source connectors and even with the deprecated `delete.handling​.mode`  option in ExtractNewRecordState SMT will cause 2 messages for a single delete event happening in the source datastore (one Debezium DELETE change event and one tombstone), depending on those configs. The implementation I am working on will de-duplicate those events so this is only temporary an issue. The reason for the previous code was that we avoided duplicate delete events for the same row/source event. But it excludes deletes for source connectors using ExtractNewRecordState SMT that drops the Debezium DELETE events and keeps tombstones (which might be semantically the best way to represent deletes in Kafka imho).

@jpechane As this code is moved to the AbstractChangeEventSink eventually with my current work, we should maybe evaluate if we close this PR as those lines will be dropped soon?

Reply:
@msillence After writing the above, maybe we can just drop those lines here for `if (record.isTombstone()) {...}` entirely and keep the  `if (record.isDelete()) { ... }` below as it handles tombstones and Debezium DELETE change event messages, yielding the previously mentioned duplicate deletes issued to the destination datastore. The idea was to skip those duplicates delete events in the default scenarion but excludes for certain ExtractNewRecordState SMT configs (which imho are even syntactically and semantically good and correct)!